src/reader/mod.rs

```diff
@@ -250,6 +250,25 @@ macro_rules! read_event_impl {
                     $self.state.state = ParseState::InsideText;
                     continue;
                 },
+                ParseState::InsideRef => { // Go to InsideText
+                    let start = $self.state.offset;
+                    match $reader
+                        .read_bytes_until(b';', $buf, &mut $self.state.offset)
+                        $(.$await)?
+                    {
+                        Ok((bytes, true)) => {
+                            $self.state.state = ParseState::InsideText;
+                            Ok(Event::GeneralRef(BytesRef::wrap(bytes, $self.decoder())))
+                        },
+                        Ok((_, false)) => {
+                            // We want to report error at `&`, but offset was increased,
+                            // so return it back (-1 for `&`)
+                            $self.state.last_error_offset = start - 1;
+                            Err(Error::Syntax(SyntaxError::UnclosedReference))
+                        }
```

Comment:
I found a code in ruffle in https://github.com/ruffle-rs/ruffle/pull/10471 from @Aaron1011 that will be break, because currently dangling `&` will always return `SyntaxError::UnclosedReference`. I think I should add a new configuration option for this here